File: C++/Level.2/큰수만들기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    int len = number.length();
    int start = 0;
    for(int i = 1; i <= len - k; ++i) {
        char max_temp = '0';
        for(int j = start; j < min(len, k + i); ++j) {
            if(max_temp < number[j]) {
                max_temp = number[j];
                start = j + 1;
            }
        }
        answer += max_temp;
    }
    return answer;
}

File: JavaScript/Level.2/큰수만들기.js
function solution(number, k) {
    let arr = [];
    let n = number.length;
    for (let i = 0; i < n; ++i) {
        while (arr.length > 0 && arr[arr.length - 1] < number[i] && k > 0) {
            arr.pop();
            --k;
        }
        arr.push(number[i]);
    }
    while (k > 0) {
        arr.pop();
        --k;
    }
    let answer = arr.join("");
    return answer;
}
